Router.d.ts

```diff
@@ -0,0 +1,8 @@
+export interface RouteDetail {
+    route: string | RegExp;
+    location: string;
+    querystring: string;
+    userData?: {};
+    component?: {};
+    name?: string;
+}
```

Comment:
Sadly, the problem is that when you add a `.d.ts` file, you need to document the types for all exports, or editors will miss the JSDoc comments.

In this file, there are some definitions missing:

- `push`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L94
- `pop`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L112
- `replace`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L125
- `link`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L158

Also, I don't know if types need to be exported for Svelte stores too? This component is exporting some Svelte stores:

- `loc`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L55
- `location`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L75
- `querystring`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L83

What about props? Do those need to be in the typing file too?

- `routes`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L236
- `prefix`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L241
- `restoreScrollState`: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L247

Reply:
I don't really know,
So router. D.ts will be generated by svelte2tsx